src/gretel_trainer/models.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from typing import Union
+
+from gretel_client.projects.models import read_model_config
+
+
+class _BaseConfig:
+    """This class should not be used directly, models should be
+    derived from this class
+    """
+
+    _max_header_clusters: int
+    """This should be overridden on concrete classes"""
+
+    # Default values for all configs, may be overridden if needed
+    _max_rows: int = 1000000
+    _enable_privacy_filters: bool = False
+
+    # Should be set by concrete constructors
+    config: Union[str, dict]
+    max_rows: int
+    max_header_clusters: int
+
+    def __init__(
+        self,
+        config: Union[str, dict],
+        max_rows: int,
+        max_header_clusters: int,
+        enable_privacy_filters: bool,
+    ):
+        self.config = read_model_config(config)
+        self.max_rows = max_rows
+        self.max_header_clusters = max_header_clusters
+        self.enable_privacy_filters = enable_privacy_filters
+
+        if not self.enable_privacy_filters:
+            logging.warning("Privacy filters disabled. Enable with the `enable_privacy_filters` param.")
+            self.update_params({"outliers": None, "similarity": None})
+
+        self.validate()
+
+    def update_params(self, params:dict):
```

Comment:
nit: `params: dict` (missing a whitespace) - also a future improvement here would to make sure that the config we're operating on is "fully qualified", if someone provided a config that didn't have every possible field on it, updating params may have no effect, is that right?

Reply:
This is a good call. I'll create an issue for the next release. We should probably add some logic to validate that the passed-in config is valid for the model being called. It should be pretty simple- we could either validate all config options (more work), or just verify that the correct model slug (e.g. synthetics, ctgan, etc) is used in the config

Reply:
Added a simple check to confirm that the model slug matches the selected model_type. Will build a more advanced check to account for use cases above in a future release.